contracts/MerkleAirdrop.sol

```diff
@@ -1,51 +1,113 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.17;
 
-import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import {MerkleProof} from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
 /**
- * @dev A contract allow recipients to claim tokens via merkle airdrop.
+ * @dev A contract that allows recipients to claim tokens via 'merkle airdrop'.
  */
-contract MerkleAirdrop is ERC20 {
-    bytes32 public immutable merkleRoot;
+contract MerkleAirdrop is ReentrancyGuard {
+    error AirdropInActive();
+    error NotInMerkleTree();
+    error AlreadyClaimed();
+    error NotOwner();
+
+    address public owner;
+    address public sender;
+    IERC20 public token;
+    bytes32 public merkleRoot;
 
-    mapping(address => bool) public hasClaimed; // use uint8
+    mapping(address => uint8) public hasClaimed;
```

Comment:
make the state variables private and create separate external getter functions for the them, if required.